lib/src/cancelable_operation.dart

```diff
@@ -125,7 +129,11 @@ class CancelableOperation<T> {
   /// Whether this operation has been canceled before it completed.
   bool get isCanceled => _completer.isCanceled;
 
-  /// Whether this operation completed before being canceled.
+  /// Whether the [CancelableCompleter] backing this operation has been
+  /// completed.
+  ///
+  /// This does not indicate that the completer backing [value] has been
```

Comment:
```dart
/// This value being true does not imply that the [value] future has completed, 
/// but merely that it is no longer possible to [cancel] the operation.
```

Maybe this value is not that useful. I would personally prefer to not have `isCompleted` at all, or have named it `canCancel`. Probably not worth the breaking change, though.

Reply:
This raises a question - if it is not possible to cancel a `CancelableOperation.fromFuture`, why does that constructor exists? What do we get out of wrapping if we can't cancel?

Reply:
You get to pass a `Future` to something expecting a `CancelableOperation`. It's an adapter, partial because the adeptee doesn't support the full API it's being adapted for.

I would probably have allowed you to cancel until the future result was available, whether the underlying completer is completed or not, but that would require more complicated plumbing - not just completing the underlying operation/future completers with the future.